src/Type/Proxy.purs

```diff
@@ -48,11 +48,140 @@
 -- | ```
 module Type.Proxy where
 
+import Prelude
+
 -- | Value proxy for kind `Type` types.
 data Proxy a = Proxy
 
+derive instance eqProxy :: Eq (Proxy a)
+
+derive instance functorProxy :: Functor Proxy
+
+derive instance ordProxy :: Ord (Proxy a)
+
+instance applicativeProxy :: Applicative Proxy where
+  pure _ = Proxy
+
+instance applyProxy :: Apply Proxy where
+  apply _ _ = Proxy
+
+instance bindProxy :: Bind Proxy where
+  bind _ _ = Proxy
+
+instance booleanAlgebraProxy :: BooleanAlgebra (Proxy a)
+
+instance boundedProxy :: Bounded (Proxy a) where
+  bottom = Proxy
+  top = Proxy
+
+instance commutativeRingProxy :: CommutativeRing (Proxy a)
+
+instance discardProxy :: Discard (Proxy a) where
```

Comment:
So this one means we'd allow discarding values of type `Proxy a` in `do` notation. `Discard` is meant for types like `Unit` so I'm not sure that's what we want.